lib/minitest/reporters/spec_reporter.rb

```diff
@@ -18,6 +28,14 @@ def start
 
       def report
         super
+        if @print_at_bottom
+          failed_tests = tests.reject { |test| test.failures.empty? }
```

Comment:
```suggestion
          failed_test_groups = tests
                               .reject { |test| test.failures.empty? }
                               .sort_by { |test| [test_class(test).to_s, test.name] }
                               .group_by { |test| test_class(test).to_s }
```
I think we can make the output a little cleaner by sorting and grouping the test failures by class name. Then we could print the class name just once per group. For example, the current output generated by the branch in this PR might look something like this:
```
Failures and errors:
MinitestReportersTest::BadTest
  test_a                                                          FAIL (0.00s)
        Expected: 1
          Actual: 2
        /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:6:in `test_a'
Location:
         /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:5


MinitestReportersTest::BadTest
  test_b                                                          FAIL (0.00s)
        Expected false to be truthy.
        /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:10:in `test_b'
Location:
         /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:9
```
With results grouped by test class, we could print something like this instead:
```
Failures and errors:
MinitestReportersTest::BadTest
  test_a                                                          FAIL (0.00s)
        Expected: 1
          Actual: 2
        /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:6:in `test_a'
Location:
         /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:5

  test_b                                                          FAIL (0.00s)
        Expected false to be truthy.
        /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:10:in `test_b'
Location:
         /Users/bingram/Projects/minitest-reporters/test/gallery/bad_test.rb:9
```

Reply:
I updated the code to sort and group the errors, makes sense, it looks much cleaner this way, thanks! :+1: